docs/source/advanced/contrib_thunder.rst

```diff
@@ -0,0 +1,468 @@
+
+Additionally, the `clang <https://github.com/Lightning-AI/lightning-thunder/blob/main/thunder/clang/__init__.py>`_ gets updated
+
+.. code-block:: python
+   :emphasize-lines: 1-3
+   :lineno-start: 1119
+
+   @clangop()
+   def unfold(a: TensorProxy, /, dim: int, size: int, step: int) -> TensorProxy:
+       return prims.unfold(a, dim, size, step)
+
+And while this symbol directly calls into the primitive and is very simple, a more involved
+example of ``clang.slice_in_dim``
+
+.. code-block:: python
+  :lineno-start: 938
+
+  @clangop()
+  def slice_in_dim(a, start_index, limit_index, stride=1, dim=0):
+      len_dim = a.shape[dim]
+      start_index = utils.canonicalize_dim_idx(len_dim, start_index)
+      limit_index = utils.canonicalize_dim_idx(len_dim, limit_index)
+
+      # Handles the start idx being greater than the dimension length by returning
+      #   a tensor with no elements
+      if start_index >= len_dim:
+          shape = list(a.shape)
+          shape[dim] = 0
+          return full(shape, 0, device=a.device, dtype=a.dtype)
+
+      # Handles the limit idx being greater than the dimension length by clamping it
+      if limit_index >= len_dim:
+          limit_index = len_dim
+
+      # Constructs args for the slice prim
+      start_indices = [0] * a.ndim
+      limit_indices = list(a.shape)
+      strides = [1] * a.ndim
+
+      start_indices[dim] = start_index
+      limit_indices[dim] = limit_index
+      strides[dim] = stride
+
+      return prims.slice_prim(a, start_indices, limit_indices, strides)
+
+enhances ``prims.slice_prim`` by adding support for default arguments and negative dimensions, among others.
+
+Once the symbol is created, we need to tell *Thunder* how to *execute* it.
+Since we are updating the ``torch`` executor, the following lines are added to the
+`executors/torchex.py <https://github.com/Lightning-AI/lightning-thunder/blob/main/thunder/executors/torchex.py>`_ file
+
+.. code-block:: python
+   :emphasize-lines: 2
+   :lineno-start: 465
+
+   unbind = _register_torch_operation("unbind")
+   unfold = _register_torch_operation("unfold", module=torch.Tensor)
+   unsqueeze = _register_torch_operation("unsqueeze")
+
+.. code-block:: python
+   :emphasize-lines: 2
+   :lineno-start: 536
+
+   _register_implementation(prims.transpose, checker=_always_executable, execution_transform=_transpose_prim_transform)
+   _register_implementation(prims.unfold, unfold, checker=_always_executable)
+   _register_implementation(prims.view, view, checker=_always_executable)
+
+the first one registers a new symbol that is directly tied to the ``torch.Tensor.unfold``, and the second
+associates this symbol with ``prims.unfold`` upon execution unless the ``checker`` fails
+(``checker`` is useful, for example, for checking PyTorch versions).
+Having ``checker=_always_executable`` always greenlights this association, and, hence,
```

Comment:
Somewhat of an offtopic question to your PR:

Why do we run checks on the meta functions for the operators instead of in the checkers?

For instance, imagine we had 2 executors who define the same operator where the first one is stricter about what it allows. If the meta function for the first one does a `utils.check(something)` then it would not give an opportunity to the second executor. As far as I understand, if this is done with a checker then it is supported.

So when do we recommend customizing the meta function vs the checker?

Reply:
I understand this but my question is, why do we even let it claim symbols whose inputs are not going to be accepted by the meta function? AFAIK the checker also has access to the inputs.

Maybe a question for @mruberry too